modules/internal/ChapelArray.chpl

```diff
@@ -3065,9 +3065,8 @@ module ChapelArray {
 
     a._instance.dsiAssignDomain(b, lhsPrivate=false);
 
-    if _isPrivatized(a._instance) {
-      _reprivatize(a._value);
-    }
+    // MPF note: should this call _reprivatize(a._value) at this point?
+    // I think so, but I see errors if I enable it.
```

Comment:
Other tests that fail with this change:
distributions/robust/arithmetic/basics/test_array_iteration1.chpl
distributions/robust/arithmetic/basics/test_array_resize_from_domain_assignment.chpl
distributions/robust/arithmetic/basics/test_array_swap.chpl
distributions/robust/arithmetic/basics/test_domain_iteration1.chpl
distributions/robust/arithmetic/basics/test_zipper_default.chpl
distributions/robust/arithmetic/composite/test_composite1.chpl
distributions/robust/arithmetic/composite/test_composite2.chpl
distributions/robust/arithmetic/kernels/hpl.chpl
distributions/robust/arithmetic/modules/test_module_Norm.chpl
distributions/robust/arithmetic/modules/test_module_Random.chpl
distributions/robust/arithmetic/modules/test_module_Search.chpl
distributions/robust/arithmetic/reindexing/test_array_alias4.chpl
distributions/robust/arithmetic/reindexing/test_reindexing4.chpl
distributions/robust/arithmetic/reindexing/test_strided_reindex1.chpl
distributions/robust/arithmetic/slicing/test_strided_slice1.chpl
distributions/robust/arithmetic/slicing/test_strided_slice2.chpl